tools/online_editor/src/index.ts

```diff
@@ -120,7 +171,92 @@ function create_share_menu(editor: EditorWidget): Menu {
             navigator.clipboard.writeText(share_url);
         },
     });
+    commands.addCommand("slint:create_gist", {
+        label: "Export to github Gist",
+        iconClass: "fa-brands fa-github",
+        mnemonic: 1,
+        isEnabled: () => {
+            return localStorage.getItem(local_storage_key_github_token) != null;
+        },
+        execute: async () => {
+            const access_token = localStorage.getItem(
+                local_storage_key_github_token,
+            );
+            console.assert(access_token != null);
+
+            // collect data:
+            const files: { [key: string]: { [key: string]: string } } = {};
+            const urls = editor.open_document_urls;
+            if (urls.length == 0) {
+                return;
+            }
+
+            const to_strip = urls[0].lastIndexOf("/") + 1;
 
+            for (const u of urls) {
+                files[u.slice(to_strip)] = {
+                    content: editor.document_contents(u) ?? "",
+                };
+            }
+
+            let extras: { [path: string]: string } = {};
+            Object.entries(editor.extra_files).forEach(async ([f, u]) => {
+                extras[f.slice(1)] = u;
+            });
+
+            const project_data = {
+                main: urls[0].slice(to_strip),
+                mappings: extras,
+            };
+
+            files["slint.json"] = { content: JSON.stringify(project_data) };
+
+            const data = JSON.stringify({
+                description:
+                    'Made with Slint, main file is: "' +
+                    urls[0].slice(to_strip) +
+                    '".',
+                public: true,
+                files: files,
+            });
+
+            fetch("https://api.github.com/gists", {
+                method: "POST",
+                mode: "cors",
+                cache: "no-cache",
+                credentials: "same-origin",
+                headers: {
+                    "Content-Type": "application/json",
+                    Accept: "application/vnd.github+json",
+                    Authorization: "Bearer " + access_token,
+                },
+                redirect: "follow",
+                referrerPolicy: "no-referrer",
+                body: data,
+            })
+                .then((response) => response.json())
+                .then((body) => {
+                    if (body.errors) {
+                        alert(
+                            "Failed to publish to Github with errors:\n" +
+                                body.errors.join("\n"),
+                        );
+                    } else if (body.html_url == null) {
+                        alert(
+                            "Failed to retrieve URL after publishing to Github",
+                        );
+                    } else {
+                        navigator.clipboard.writeText(body.html_url);
```

Comment:
there should be some sort of feedback that this happened.

Also, i'm not a fan of such interface that mess with my clipboard, i'd prefer if this opened a popup with the url so i can click on it to open it or copy it myself.  But maybe that's just me.